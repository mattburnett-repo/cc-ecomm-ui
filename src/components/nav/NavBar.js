export default function NavBar (props) {
    const { calledFrom } = props;

    return (
        <div>
            <div role="presentation" aria-label="header-message">header message</div>
            if(calledFrom !== 'landing') {
                <>
                    <button aria-label="home">Home</button>
                    <button aria-label="show-cart">Show Cart</button>
                </>
            };
            if(calledFrom === 'cart') {
                <button aria-label="checkout">Checkout</button>
            }
            if(calledFrom === 'shippingInfo' || calledFrom === 'paymentInfo') {
                <button aria-label="cancel">Cancel</button>
            }
            <button aria-label="logout">Log Out</button>
        </div>
    )
}