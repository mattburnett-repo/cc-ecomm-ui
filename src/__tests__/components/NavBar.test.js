// 20211217: any wireframe will do...

import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';

import NavBar from '../../components/nav/NavBar';

describe('<NavBar /> component tests', () => {
    beforeEach(() => render(<NavBar />));

    it('should display a header message', () => { // TODO: check other components that have header messages, and refactor to here
        screen.getByRole('presentation', {name: /header-message/i});
    });
    it('should display a logout button', () => {
        screen.getByRole('button', {name: /logout/i});
    });
}); // end component

describe('<NavBar /> feature tests', () => {
    it('renders a snapshot', () => {
        const tree = renderer.create(<NavBar />).toJSON();
        expect(tree).toMatchSnapshot();
    });
}); // end feature

describe('<NavBar /> unit / functional tests', () => {
    it('displays and clicks a home button if user is not on landing page', () => {
        render(<NavBar calledFrom="login" />);
        
        const theVal = screen.getByRole('button', { name: /home/i});
        fireEvent.click(theVal);
    });

    it('displays and clicks a show cart button if user is not on landing page', () => {
        render(<NavBar calledFrom="login" />);

        const theVal = screen.getByRole('button', { name: /show-cart/i});
        fireEvent.click(theVal);
    });

    it('displays and clicks a checkout button if user is on the cart page', () => {
        render(<NavBar calledFrom="cart" />);

        const theVal = screen.getByRole('button', { name: /show-cart/i});
        fireEvent.click(theVal);
    });

    it('displays and clicks a cancel button if user is on the shipping info page', () => {
        render(<NavBar calledFrom="shippingInfo" />);

        const theVal = screen.getByRole('button', { name: /cancel/i});
        fireEvent.click(theVal);
    });
    it('displays and clicks a cancel button if user is on the payment info page', () => {
        render(<NavBar calledFrom="paymentInfo" />);

        const theVal = screen.getByRole('button', { name: /cancel/i});
        fireEvent.click(theVal);
    });
}); // end component