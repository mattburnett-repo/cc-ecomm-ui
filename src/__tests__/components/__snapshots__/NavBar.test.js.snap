// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> feature tests renders a snapshot 1`] = `
<div
  aria-label="nav-bar"
  role="presentation"
>
  <div
    aria-label="header-message"
    role="presentation"
  >
    header message
  </div>
  if(
   !== 'landing') 
  <button
    aria-label="home"
  >
    Home
  </button>
  <button
    aria-label="show-cart"
  >
    Show Cart
  </button>
  ; if(
   === 'cart') 
  <button
    aria-label="checkout"
  >
    Checkout
  </button>
  if(
   === 'shippingInfo' || calledFrom === 'paymentInfo') 
  <button
    aria-label="cancel"
  >
    Cancel
  </button>
  <button
    aria-label="logout"
  >
    Log Out
  </button>
</div>
`;
